internal/service/ec2/ec2_ami.go

```diff
@@ -277,6 +277,10 @@ func resourceAMI() *schema.Resource {
 				ForceNew:         true,
 				ValidateDiagFunc: enum.Validate[awstypes.TpmSupportValues](),
 			},
+			"uefi_data": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
```suggestion
				Optional: true,
				ForceNew: true,
```

Needs _ForceNew_ as this attribute can't be updated.